### PR TITLE
Feature/33373 csv import tickets

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/33373

This PR adds support for tickets import in the CORE plugin

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Whitespace found at end of line | [Admin_Page.php:163](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L163)
:x: | Whitespace found at end of line | [Admin_Page.php:165](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L165)
:x: | Whitespace found at end of line | [Admin_Page.php:168](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L168)
:x: | There must be a single space after the first parenthesis | [Admin_Page.php:171](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L171)
:x: | There must be a single space before the last parenthesis | [Admin_Page.php:171](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L171)
:x: | There must be a single space after the first parenthesis | [Admin_Page.php:172](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L172)
:x: | There must be a single space before the last parenthesis | [Admin_Page.php:172](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L172)
:x: | There must be a single space after the first square bracket | [Admin_Page.php:172](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L172)
:x: | There must be a single space before the last square bracket | [Admin_Page.php:172](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L172)
:x: | There must be a single space after the first square bracket | [Admin_Page.php:172](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L172)
:x: | There must be a single space before the last square bracket | [Admin_Page.php:172](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/Tribe/Importer/Admin_Page.php#L172)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$import_type_title' | [columns.php:22](https://github.com/moderntribe/the-events-calendar/blob/4a3261be333d269fd79b7bd2cfad8063a3ea5a1f/src/io/csv/admin-views/columns.php#L22)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Importer/Admin_Page.php [hunk 1]
```diff
@@ -160,6 +160,16 @@ public function render_csv_tab() {
...
+	
+				$import_type_titles_map = array();
+				
+				/**
+				 * Allows filtering the import type titles to go from a slug to a pretty title.
+				 * 
+				 * @param array $import_type_titles_map
+				 */
+				$import_type_titles_map = apply_filters('tribe_events_import_type_titles_map', $import_type_titles_map);
+				$import_type_title = isset($import_type_titles_map[$import_type]) ? $import_type_titles_map[$import_type] : ucwords( $import_type );
```
Comment (barryhughes) [inline src/Tribe/Importer/Admin_Page.php]: Sorry to nitpick, but, spacing in the above two lines could do with a tweak (add spaces after opening brackets and before closing them etc) :grin:

File: src/Tribe/Importer/File_Importer.php [hunk 2]
```diff
@@ -114,7 +126,7 @@ public function import_next_row( $throw = false ) {
 		$row    = $this->reader->get_last_line_number_read() + 1;
 		if ( ! $this->is_valid_record( $record ) ) {
 			if ( ! $throw ) {
-				$this->log[ $row ] = sprintf( esc_html__( 'Missing required fields in row %d.', 'the-events-calendar', $row ) );
+				$this->log[ $row ] =$this->get_skipped_row_message($row);
```
Comment (barryhughes) [inline src/Tribe/Importer/File_Importer.php]: Add a space after `=`

**Comment by barryhughes** [comment]
Looks good! Tagged enhance re some formatting items (as picked up by tr1b0t), otherwise good to merge :+1:

**Comment by bordoni** [comment]
Can we delete the branch here? @lucatume

**Comment by lucatume** [comment]
Sure! My bad, thanks man.

**Pull request merged**